java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/RelationalSymbolicValue.java

```diff
@@ -197,41 +205,15 @@ RelationalSymbolicValue inverse() {
     return states;
   }
 
-  private boolean checkRelation(BooleanConstraint booleanConstraint, ProgramState programState) {
-    RelationState relationState = resolveState(programState);
-    return !relationState.rejects(booleanConstraint);
-  }
-
-  RelationState resolveState(ProgramState programState) {
+  RelationState resolveState(Set<RelationalSymbolicValue> knownRelations) {
```

Comment:
lack VisibleForTesting annotation.
I know this vocabulary does not come from you but name of the method is confusing : resolveRelationState would be a bit more explicit (overall RelationState is a bad name IMO but let's not change it now)